manager/src/main/java/io/gingersnapproject/Maps.java

```diff
@@ -0,0 +1,45 @@
+package io.gingersnapproject;
+
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Stream;
+
+import javax.inject.Singleton;
+
+import org.infinispan.commons.marshall.WrappedByteArray;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+@Singleton
+public class Maps {
```

Comment:
Sure, I only tossed this together quickly to get something working.

Reply:
Actually I am not liking how Rules/Rule is working out with the code you suggested. 
I will however rename it to Rules.

Reply:
And actually the more I think about the usage with the database, I think we will want to have an memory service such as this and a database service. The latter will have to be non blocking, so we don't want to tie their APIs together to prevent allocations. And then the CommandProcessor will handle the iteraction between the two I am thinking.

Reply:
I renamed it to Caches for this PR for now. I am not a fan of the abstractions this is doing currently, but I want to just get this in first and the iterate on it to be honest.